### PR TITLE
fix(SFI-1657): handle empty installments configuration

### PR DESCRIPTION
**Pull request by shanikantsingh**
## Description

Fixes a critical checkout regression where **no payment methods render** for the
installment locales (`pt_BR`, `ja_JP`, `tr_TR`, `es_MX`) whenever credit card
installments are **disabled** in the Business Manager.

### Root cause

When installments are disabled, `getAdyenViewData()` in `begin.js` returned a JS
object (`{}`) for `installments`. The billing ISML renders this into a
single-quoted string literal (`window.installments = '${pdict.adyen.installments}'`),
so the object stringifies to `[object Object]`, producing
`window.installments = '[object Object]'`.

For the installment locales, `cardConfig.js` calls
`JSON.parse(window.installments...)` without any error handling. Parsing
`[object Object]` throws a `SyntaxError`, which propagates up through
`getConfig()` and takes down the entire payment-methods configuration — so
nothing renders. Non-installment locales are unaffected because
`setInstallments()` returns early before the parse.

This was introduced when the installments logic moved into `cardConfig.js`
during the adyen-web v6 modernization, which dropped the previous swallowing
`try/catch`.

## Changes

- **`begin.js`** — return a valid empty-array JSON string (`'[]'`) instead of an
  object in the disabled branch, so `window.installments` parses to `[]` and is
  safely skipped.
- **`cardConfig.js`** — wrap the `JSON.parse` in a `try/catch`. On any
  malformed/empty/undefined value, bail out of `setInstallments()` without
  throwing so `getConfig()` always returns a valid config. This also covers the
  enabled-but-empty-preference edge case (`JSON.parse('')`).

## Test

- [x] Unit test asserting `installments === '[]'` when installments are disabled.
- [x] Manual test with installments disabled, verify payment methods render for `pt_BR`, `ja_JP`, `tr_TR`, `es_MX`.
- [x] Manual test with installments enabled, verify installment options still render correctly for those locales.

**Fixed issue**:  #SFI-1657

### CODE REVIEW

File: src/cartridges/int_adyen_SFRA/cartridge/controllers/middlewares/checkout/begin.js [hunk 1]
```diff
@@ -43,7 +43,7 @@ function getAdyenViewData() {
   const environment = AdyenHelper.getCheckoutEnvironment();
   const installments = AdyenConfigs.getAdyenInstallmentsEnabled()
     ? AdyenConfigs.getCreditCardInstallments()
-    : {};
+    : '[]';
```
[the remark this answers is not in the document]
Comment (shanikantsingh) [reply src/cartridges/int_adyen_SFRA/cartridge/controllers/middlewares/checkout/begin.js:46]: fixed

**Review by amihajlovski — APPROVED** [review]

**Review by shubhamk67 — APPROVED** [review]
